src/router/link.ts

```diff
@@ -1,36 +1,267 @@
+    return matches[matches.length - 1]?.fullPath;
+  });
 
   private navigateOptions = computed(() => {
-    const to = this.toOptions();
-    if (typeof to === 'object') return to;
-    return { to };
+    return { ...this.linkOptions(), from: this.from() };
   });
 
-  protected hostHref = computed(
-    () => this.router.buildLocation(this.navigateOptions()).href
-  );
+  private next = computed(() => {
+    const [options] = [this.navigateOptions(), this.currentSearch()];
+    return this.router.buildLocation(options);
+  });
+
+  private preload = computed(() => {
+    const userReloadDocument = this.userReloadDocument();
+    if (userReloadDocument) return false;
+    const userPreload = this.userPreload();
+    if (userPreload) return userPreload;
+    return this.router.options.defaultPreload;
+  });
+
+  private preloadDelay = computed(() => {
+    const userPreloadDelay = this.userPreloadDelay();
+    if (userPreloadDelay) return userPreloadDelay;
+    return this.router.options.defaultPreloadDelay;
+  });
+
+  protected hostHref = computed(() => {
+    const [type, to] = [this.type(), this.to()];
+    if (type === 'external') return to;
+
+    const disabled = this.disabled();
+    if (disabled) return undefined;
+
+    const next = this.next();
+    return next.maskedLocation
+      ? this.router.history.createHref(next.maskedLocation.href)
+      : this.router.history.createHref(next.href);
+  });
+
+  private ready = signal(false);
+  transitioning = signal(false);
+  isActive = computed(() => {
+    if (!this.ready()) return false;
 
-  navigate($event: Event) {
-    $event.preventDefault();
+    const [next, location, exact] = [
+      this.next(),
+      this.location(),
+      this.exactActiveOptions(),
+    ];
+    if (exact) {
+      const testExact = exactPathTest(
+        location.pathname,
+        next.pathname,
+        this.router.basepath
+      );
+      if (!testExact) return false;
+    } else {
+      const currentPathSplit = removeTrailingSlash(
+        location.pathname,
+        this.router.basepath
+      ).split('/');
+      const nextPathSplit = removeTrailingSlash(
+        next.pathname,
+        this.router.basepath
+      ).split('/');
+      const pathIsFuzzyEqual = nextPathSplit.every(
+        (d, i) => d === currentPathSplit[i]
+      );
+      if (!pathIsFuzzyEqual) {
+        return false;
+      }
+    }
+
+    const includeSearch = this.includeSearchActiveOptions() ?? true;
+
+    if (includeSearch) {
+      const searchTest = deepEqual(location.search, next.search, {
+        partial: !exact,
+        ignoreUndefined: !this.explicitUndefinedActiveOptions(),
+      });
+      if (!searchTest) {
+        return false;
+      }
+    }
+
+    const includeHash = this.includeHashActiveOptions();
+    if (includeHash) {
+      return location.hash === next.hash;
+    }
+
+    return true;
+  });
+
+  constructor() {
+    effect(() => {
+      this.ready.set(true);
```

Comment:
This is a hack to have `isActive()` working correctly. If we access `link.isActive()` on the template for `[class.active]` for example, it'd be too early for input to resolve resulting in the `input.required` throwing

Reply:
use `linkActive` input instead